pkg/sfu/downtrack.go

```diff
@@ -202,6 +202,9 @@ type DownTrack struct {
 	// max layer change callback
 	onSubscribedLayersChanged func(dt *DownTrack, layers VideoLayers)
 
+	// target layer found callback
+	onTargetLayerFound func(dt *DownTrack)
```

Comment:
Sending resume when we actually lock to a target layer and start forwarding rather than when an allocation is done. This aligns better with media flow. Using this callback into streamallocator and streamallocator will check if a previous `pause` was sent before sending an `active` for this stream.